src/index.js

```diff
@@ -37,36 +35,33 @@ Cache.prototype.upsert = function (key, geojson, options = {}, callback = noop)
 }
 
 Cache.prototype.update = function (key, geojson, options = {}, callback = noop) {
-  // support a feature collection or an array of features
   if (!this.store.has(key)) return callback(new Error('Resource not found'))
-  const features = geojson.features ? geojson.features : geojson
-  this.store.set(key, features)
+  this.store.set(key, geojson)
   const existingMetadata = this.catalog.store.get(key)
   const metadata = geojson.metadata || existingMetadata
   if (options.ttl) metadata.expires = Date.now() + (options.ttl * 1000)
   this.catalog.update(key, metadata, callback)
 }
 
 Cache.prototype.append = function (key, geojson, options = {}, callback = noop) {
-  const features = geojson.features ? geojson.features : geojson
+  const features = geojson.features ? geojson.features : []
```

Comment:
Ahh, I didn't realise certain implementations input only an array of features. I'll make sure to change that. Good idea to make it a helper function. 

I will also double check the impact of this on the other functions. To be sure: we can assume that (from now on at least) it is always a geojson being stored `koop-cache-memory` right? It's just the input that could be an array only? 

Thanks @rgwozdz, I will look at it soon.

Reply:
Yes, it seems that the `insert`, `update`, and `append` methods all allow either a geojson object or a feature array as input.

For the `append` we want to get to the feature array and merge it into the cache array. But for `update` and `insert`, we want to cache a geojson object.  So if a feature array is passed in, we should wrap it in a GeoJSON object (just like the old `retrieve` method did before returning it). 

> To be sure: we can assume that (from now on at least) it is always a geojson being stored koop-cache-memory right?

Yes, I think that's best for several reasons: 1) to preserve metadata not stored in the `metadata` wrapper (e.g. GeoJSON's `crs` property), and 2) to prevent us from re-wrapping the features array in GeoJSON on every cache-read event. Doing so will slightly increase the memory used by the cache but overall it's a small amount relative to that occupied by the feature array.

Reply:
Hi @rgwozdz !

I committed the helper that we discussed above, so we should be sure it's always a geojson. Also added some tests. Note that I only added a test for the helper to do a small check if the conversions are done correctly.

What I was wondering though; shouldn't we just throw an error when the geojson is not what we expect (i.e. when it's empty)? But I'm also not sure if that's easy to keep tidy with a(nother) helper /validation function, as it also needs to callback the error then. So for now just this commit.

Guus